lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -301,4 +301,334 @@ protected function get_block_classes( $style_nodes ) {
+			'properties'        => array( 'padding', 'margin' ),
+		),
+	);
+
+	/**
+	 * The valid properties under the settings key.
+	 *
+	 * @var array
+	 */
+	const VALID_SETTINGS = array(
+		'appearanceTools' => null,
+		'border'          => array(
+			'color'  => null,
+			'radius' => null,
+			'style'  => null,
+			'width'  => null,
+		),
+		'color'           => array(
+			'background'       => null,
+			'custom'           => null,
+			'customDuotone'    => null,
+			'customGradient'   => null,
+			'defaultGradients' => null,
+			'defaultPalette'   => null,
+			'duotone'          => null,
+			'gradients'        => null,
+			'link'             => null,
+			'palette'          => null,
+			'text'             => null,
+		),
+		'custom'          => null,
+		'layout'          => array(
+			'contentSize' => null,
+			'wideSize'    => null,
+		),
+		'spacing'         => array(
+			'customSpacingSize' => null,
+			'spacingSizes'      => null,
+			'spacingScale'      => null,
+			'blockGap'          => null,
+			'margin'            => null,
+			'padding'           => null,
+			'units'             => null,
+		),
+		'typography'      => array(
+			'customFontSize' => null,
+			'dropCap'        => null,
+			'fontFamilies'   => null,
+			'fontSizes'      => null,
+			'fontStyle'      => null,
+			'fontWeight'     => null,
+			'letterSpacing'  => null,
+			'lineHeight'     => null,
+			'textDecoration' => null,
+			'textTransform'  => null,
+		),
+	);
+
+	/**
+	 * Given the block settings, it extracts the CSS Custom Properties
+	 * for the presets and adds them to the $declarations array
+	 * following the format:
+	 *
+	 * ```php
+	 * array(
+	 *   'name'  => 'property_name',
+	 *   'value' => 'property_value,
+	 * )
+	 * ```
+	 *
+	 * @param array $settings Settings to process.
+	 * @param array $origins  List of origins to process.
+	 * @return array Returns the modified $declarations.
+	 */
+	protected static function compute_preset_vars( $settings, $origins ) {
+		$declarations = array();
+
+		foreach ( static::PRESETS_METADATA as $preset_metadata ) {
+			$values_by_slug = static::get_settings_values_by_slug( $settings, $preset_metadata, $origins );
+
+			foreach ( $values_by_slug as $slug => $value ) {
+				$declarations[] = array(
+					'name'  => static::replace_slug_in_string( $preset_metadata['css_vars'], $slug ),
+					'value' => $value,
+				);
+			}
+		}
+
+		return $declarations;
+	}
+
+	/**
+	 * Transform the spacing scale values into an array of spacing scale presets.
+	 *
+	 * @param array $spacing_scale scale Array of values used to create the scale of spacing presets.
+	 * @return array An array of spacing preset sizes.
+	 */
+	protected static function get_spacing_sizes( $spacing_scale ) {
+		$spacing_sizes   = array();
+		$spacing_sizes[] = array(
+			'name' => 0,
+			'slug' => 0,
+			'size' => 0,
+		);
+		$spacing_sizes[] = array(
+			'name' => 1,
+			'slug' => 10,
+			'size' => $spacing_scale['firstStep'] . $spacing_scale['units'],
```

Comment:
Should we add validation / checks for each of these properties at the top of this function? What happens if someone accidentally enters an illegal value for any of these properties? (E.g. if `steps` is a string)

Reply:
I'm also seeing a bunch of ` Undefined index` notices for `$spacing_scale['firstStep']`, `$spacing_scale['units']` and `$spacing_scale['steps']`

Reply:
I have added some checks for valid values, and triggering a notice if some of the set values are invalid. Another approach would be to fall back to core defaults if any of the settings are invalid ... what do you think is the best approach?